routes/apply_discount.js

```diff
@@ -232,6 +234,65 @@ const responseSchema = {
   'type': 'object',
   'additionalProperties': false,
   'properties': {
+    'buy_together': {
+      'type': 'array',
+      'maxItems': 20,
+      'product_kit_discounts': {
+        'type': 'object',
+        'additionalProperties': false,
+        'maxProperties': 50,
+        'patternProperties': {
+          '^([a-f0-9]{24})$': {
+            'type': 'object',
+            'additionalProperties': false,
+            'properties': {
+              'discount': {
+                'type': 'object',
+                'required': [ 'value' ],
+                'additionalProperties': false,
+                'properties': {
+                  'type': {
+                    'type': 'string',
+                    'enum': [ 'percentage', 'fixed' ],
+                    'default': 'percentage',
+                    'description': 'Discount type'
+                  },
+                  'value': {
+                    'type': 'number',
+                    // 'multipleOf': 0.0001,
+                    'minimum': -99999999,
+                    'maximum': 99999999,
+                    'description': 'Discount value, percentage or fixed'
+                  }
+                },
+                'description': 'Discount object'
+              },
+              'items': {
+                'type': 'array',
+                'maxItems': 3000,
+                'items': {
+                  'type': 'object',
+                  'additionalProperties': false,
+                  'required': [ 'product_id' ],
+                  'properties': {
+                    'product_id': {
+                      'type': 'string',
+                      'pattern': '^[a-f0-9]{24}$',
+                      'description': 'Product ID'
+                    }
+                  },
+                  'description': 'One of the products kit'
+                },
+                'description': 'Products composing kit'
```

Comment:
`{ items: [ { product_id: 1 }, { product_id: 2 } ]` pra quê essa complicação? 😬 
Por que não só `{ product_ids: [1, 2] }`?

Reply:
Além disso, até pode ser necessário ter um array de IDs de produtos se (e somente se) for a intenção suportar compre junto com mais de 2 produto (o produto atual + 2), o app de descontos já faz isso?
Nesse caso, se vai ter um array com com product IDs não faz sentido que a propriedade do objeto seja o ID do produto (pattern property com regex pra obejct ID), você tá pedindo pro app retornar o ID do produtos em 2 lugares...

Reply:
Se for compre junto com 2 produtos apenas (o atual + 1) não tem necessidade de array com product IDs, aí mantém o ID como propriedade do objeto via pattern properties.

Reply:
O compre junto do app, suporta quantos itens eu colocar lá. Inclusive, se não colocar nenhum id de produto, gera o compre junto pra loja inteira, porque antigamente colocar todos os ids no app dava pau.